### PR TITLE
Update Migration to TS guidance

### PR DESCRIPTION
**Pull request by Darth-Knoppix**
As discussed in Frontend Guild, migration to TS can be difficult when relying on JS components. This will help alleviate compilation errors due to untyped code during the migration.

### CODE REVIEW

File: languages/JavaScript/index.md [hunk 1]
```diff
@@ -27,6 +27,27 @@ There are ways to avoid typechecking when using TypeScript such as `Any` and `@t
...
+
+A quick fix is to add a `/* @ts-ignore */` directive above the component usage like so (assumes `PrimaryPrice` is a JS component):
+
+```jsx
+<Spacer height={unit(2)} mobileHeight={unit(3)} />
+{/* @ts-ignore */}
+<PrimaryPrice price={price} />
+```
+
+This approach is preferred when it's not feasible to change the components you're importing. When a component fincally makes the transition to TS, any `/* @ts-ignore */` directives should be removed where this new TS component is referenced.
```
Comment (shoshana123) [inline languages/JavaScript/index.md:49]: I think there might be a small typo for `finally`

**Review by shoshana123 — APPROVED** [review]
Looks good. Just a small note to maybe double check?

File: languages/JavaScript/index.md (hunk 1 above)
Comment (Darth-Knoppix) [reply languages/JavaScript/index.md]: Nice catch, thanks!

**Review by martyphee — COMMENTED** [review]
Ship it

**Pull request merged**